articles/151_node_to_participant_mapping.md

```diff
@@ -0,0 +1,241 @@
+For that reason, the QoS of the `Publishers` should be:
+
+- Durability: Transient Local
+- History: Keep Last
+- History depth: 1
+- Reliability: Reliable
+
+Considering that a keyed topic will be used, in which the history depth apply for each key, only one `Publisher` per process will be needed.
+In that case, `unregister_instance` can be used for delete that key from the history (see [RTI Managing Data Instances](https://community.rti.com/static/documentation/connext-dds/5.2.3/doc/manuals/connext_dds/html_files/RTI_ConnextDDS_CoreLibraries_UsersManual/Content/UsersManual/Managing_Data_Instances__Working_with_Ke.htm)).
+
+The QoS of the `Subscriber` should be:
+
+- Durability: Transient Local
+- History: Keep Last
+- History depth: 1
+- Reliability: Reliable
+
+The subscriber could access data in two different ways:
+- Polled and accessed using `Subscriber` read method when needed.
+- Listened, accessed using subscriber take method and organized in a local cache.
+
+The second option allows better organization of this information (e.g.: in hash tables).
+
+#### Using USER_DATA and GROUP_DATA QoSPolicy
+
+Each `Participant` could have in its user data the list of node names that owns.
+When this data is changed, each `ParticipantListener` will be notified.
+This is not a good option, as `UserData` is just a sequence of bytes.
+Organizing a complex message in it won't be easy nor performant.
+
+Similarly to `UserData`, `GroupData` is a available in `Publishers` and `Subscribers`.
+These entities only need to communicate the GUID of the `Participant` and the `Node` name from which it was created.
+This idea can be combined with a topic just publishing the list of `Node` names, without including all the other vectors in the message.
+Although, it is more difficult to communicate this information for `Services` and `Clients`, as they use behind the scenes just a `DDS Publisher` and `Subscriber`.
+
+### Which layer will be implemented in?
+
+The implementation can be done in two different ways:
+
+- Through rmw implementations to rcl.
+- Modify rmw implementations without modifying rmw API (as long as possible).
+
+The first approach have the following disadvantages:
+- There is no `Node` concept in `rmw` layer, as `Node` discovery is solved in `rcl`.
+  Actually, all the `Node` related API in `rmw` doesn't have more sense.
+- Currently, no threads are created in the `rcl` layer.
+  It will be needed in case `Node` discovery is done in this layer.
+- It will force to build the concept of `Node` on top of the underlying middleware in use, regardless if the middleware already has a lightweight entity similar to a `Node`.
+- It will break API in many layers.
+
+
+The second approach has the following disadvantages:
+- Each `rmw` have to reimplement node discovery logic.
+  - It can be worked arround creating a new common package that uses the abstractions in `rmw`.
```

Comment:
This doesn't sound like a second disadvantage, but a continuation of the first.
```suggestion
  This can be avoided by creating a common package that uses the abstractions in `rmw`.
```